### ISSUE TITLE
Column ordering doesn't work

### ISSUE THREAD

**Issue by geoffs3310** [yajra/laravel-datatables#63]
Clicking a column heading to order the column has no effect. The correct ordering parameters get sent through to the server but the data returned is in the same order.

**Comment by haytamkh** [comment]
version update fixed same issue , just update composer.json to v5 and do composer update

**Comment by yajra** [comment]
@geoffs3310 what version are you using? Anyways, I suggest you check you js and make sure you have defined your column definitions properly. Sorting/searching feature are dependent on `columns.name` defined on your js. See example code below for reference.

```js
    $('#users-table').DataTable({
        processing: true,
        serverSide: true,
        ajax: '/users-data',
        search: {
            caseInsensitive: true
        },
        columns: [
            {data: 'id', name: 'id'},
            {data: 'name', name: 'name'},
            {data: 'email', name: 'email'},
            {data: 'created_at', name: 'created_at'},
            {data: 'updated_at', name: 'updated_at'}
        ]
    });
```

**Comment by zeckdude** [comment]
I'm using version 3.6.4 and I'm experiencing the same problem. I add the columns and I'm getting the following error: 

>  DataTables warning: table id=test_hotels - Requested unknown parameter 'name' for row 0. For more information about this error, please see http://datatables.net/tn/4

Here's my code:

**Controller:**
```     
public function getData() {
  $hotels = Hotel::select(array('name','state'));
  return Datatables::of($hotels)->make();
}
```

**View:**
```
<table id="test_hotels" class="table table-hover table-condensed">
    <thead>
        <tr>
            <th class="col-md-3">Hotel Name</th>
            <th class="col-md-3">State</th>
        </tr>
    </thead>
</table>
```

**JS:**
```
$(document).ready(function() {
  $('#test_hotels').dataTable( {
    "serverSide": true,
    "processing": true,
    "ajax": "/hotel/data",
    "pageLength": 50,
    columns: [
      {data: 'name', name: 'name'},
      {data: 'state', name: 'state'}
    ],
    "language": {
      "lengthMenu": "Show _MENU_ hotels",
      "zeroRecords": "No matching hotels found",
      "info": "Showing _START_ to _END_ of _TOTAL_ hotels",
      "infoEmpty": "Showing 0 to 0 of 0 hotels",
      "infoFiltered": "(filtered from _MAX_ total hotels)",
      "processing": "Loading Hotels..."
    }
  });
});
```

The data loads in great and I can search, but the ordering isn't working as expected.

**Comment by yajra** [comment]
@zeckdude I'm also using Laravel 4 with v3.6.4 on my project and it's working as expected following the conventions I've provided. What version of Datatables are you using? I'm using DT1.10.7.

**Comment by zeckdude** [comment]
@yajra Thanks for the quick response! Your package is incredibly helpful so thanks for that!

I'm using DataTables 1.10.7 with and dataTables.bootstrap.js.

**Comment by zeckdude** [comment]
I'm running some tests in order to try and figure out the problem. I installed your demo site that is running Laravel 5 and it works great for me there. I am comparing the code with what I have and it looks like the same, so I'm confused.

I did notice that on your Basic Eloquent example, you didn't define the columns.name in the js, yet the sorting worked anyways. Does that mean the columns option is not required for the sorting?

**Comment by yajra** [comment]
For basic example, yes it would work without defining the columns. However, I do not advise using it because it will greatly depend on the arrangements of selected columns. That is the **OLD** way of using the package.

**Comment by yajra** [comment]
For your example codes above, it looks good and I don't find any issue on it.

**Comment by zeckdude** [comment]
That's frustrating. For the columns name and data, should those both be the name of the column in the table?

**Comment by yajra** [comment]
For **data**, it corresponds to the data object you want to use to display as it's value.
For **name**, it corresponds to the `column name` that should be used for searching/sorting in the database.

**Comment by yajra** [comment]
related to #64 for Laravel 5. Fixed on v3.6.5 for Laravel 4 on commit 5cffdb78864586189ce19cf4e187400e42ac78bc
